scripts/scripts.js

```diff
@@ -310,6 +312,26 @@ export function getWindowSize() {
     height: windowHeight,
   };
 }
+
+export function addTopSpacingStyleToFirstMatchingSection(main) {
+  const excludedClasses = ['static', 'spacer-container', 'feed-container', 'modal-fragment-container', 'hero-banner-container', 'hero-career-container', 'breadcrumb-container', 'hero-horizontal-tabs-container', 'carousel-container'];
```

Comment:
I am not able to properly understand this logic. Can you explain

Reply:
Sure
We basically look at the first 3 sections, if the section contains one of the blocks that should not have spacing at the top (carousel, hero etc; there's an example of an affected and a not affected page in the pr description) we don't do anything; if it does not contain one of the blocks we add the `.auto-top-spacing` class to it which adds `margin-top: 3rem` 
The reason why we look at the first 3 sections is that, sections that can be a candidate for this autoblocking are gonna be somewhere in these 3 first sections of a page
1. Usually the first section is the `breadcrumb` section (but not always) so we skip that
2. The second one can be a `hero` or `accordion` etc (also not always) so we skip that as well
3. The third one is usually gonna be the section that gets the `top-margin` 
4. If we reach the 4th section (index === 3) or we already added the `top-margin` we just break out of the loop to stop looking for more candidates

The reason why I used `every()` instead of `forEach()` is that we can't break out of a foreach loop, but if you `return false` in an `every()` loop its equivalent to breaking out of if, so, `return false` === `break`